### PR TITLE
drivers 802154: use memset for eui64 padding

### PR DESCRIPTION
**Pull request by OlegHahm**
As proposed by @authmillenon in https://github.com/RIOT-OS/RIOT/pull/5239#discussion_r58342950.

### CODE REVIEW

File: drivers/at86rf2xx/at86rf2xx.c [hunk 1]
```diff
@@ -69,9 +69,7 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
 
 #if CPUID_LEN < 8
     /* in case CPUID_LEN < 8, fill missing bytes with zeros */
-    for (int i = CPUID_LEN; i < 8; i++) {
-        cpuid[i] = 0;
-    }
+    memset(&(cpuid[CPUID_LEN]), 0, (IEEE802154_LONG_ADDRESS_LEN - CPUID_LEN));
```
Comment (authmillenon) [inline drivers/at86rf2xx/at86rf2xx.c]: Is it really necessary to have all this additions in here? I think the time advantage we get from this is negligent compared to the loss of readability and some byte of ROM we loose with this method here.

**Comment by kaspar030** [comment]
Feels like if we touch this, we should remove the code duplication...

**Comment by authmillenon** [comment]
> Feels like if we touch this, we should remove the code duplication...

not related to this PR, IMHO. Let's do this later down the line.

**Comment by OlegHahm** [comment]
> Feels like if we touch this, we should remove the code duplication...

Yes, I had this feeling, too, but I don't have the time to tackle this right now.

File: drivers/at86rf2xx/at86rf2xx.c (hunk 1 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: > Is it really necessary to have all this additions in here

I don't understand? Which "additions" are you talking about?

Comment (authmillenon) [inline drivers/at86rf2xx/at86rf2xx.c]: I would just write

```C
memset(cpuid, 0, CPUID_LEN);
```

I was referring to the (implicit) additions in `&cpuid[CPUID_LEN]` and `(IEEE802154_LONG_ADDRESS_LEN - CPUID_LEN)`

Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: Ah, you mean, first memset and then fill it. Makes sense.

**Comment by authmillenon** [comment]
Needs rebase.

File: drivers/at86rf2xx/at86rf2xx.c (hunk 1 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: addressed

File: drivers/at86rf2xx/at86rf2xx.c [hunk 2]
```diff
@@ -70,18 +70,17 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
     dev->netdev.flags = 0;
     /* set short and long address */
 #if CPUID_LEN
+    /* in case CPUID_LEN < 8, fill missing bytes with zeros */
+    memset(&(cpuid[CPUID_LEN]), 0, CPUID_LEN);
```
Comment (authmillenon) [inline drivers/at86rf2xx/at86rf2xx.c]: Here and below: this call let's the buffer overflow in any case. Please use either `cpuid` or `&cpuid[0]` as a starting point.

Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx.c]: Oops, something must have gone wrong here during the copy pasta.

**Comment by OlegHahm** [comment]
addressed comment

**Comment by authmillenon** [comment]
ACK, please squash

**Comment by OlegHahm** [comment]
squashed

**Comment by cgundogan** [comment]
I know it's marked for 2016.07, but would we want this backported in 2016.04?

**Comment by OlegHahm** [comment]
It is already part of the release branch.

**Pull request merged**